### PR TITLE
Remove duplicate function declarations-v3

### PR DESCRIPTION
**Pull request by Nancyenos**
Ticket: #7297

Make sure these boxes are checked accordingly before submitting your Pull Request -- thank you.

## Contribution style:
- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html

## Our Contribution agreements:
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)

Link to ticket: https://redmine.openinfosecfoundation.org/issues/7297

Describe changes:
-Remove duplicate function declarations fro header files
-I used this to find the duplicates:
```
cat src/*.h | grep -E "^[a-zA-Z_].*\(.*\);" | sort | uniq -c | awk '$1 == 2'
```
- Then to find specific files where the funtions are, I used:
```
grep -Rn "DetectEngineBufferTypeGetById" src/*.h

```
Previous pr #11990

### CODE REVIEW

File: src/detect.h [hunk 1]
```diff
@@ -1594,8 +1594,6 @@ void *DetectGetInnerTx(void *tx_ptr, AppProto alproto, AppProto engine_alproto,
 void RuleMatchCandidateTxArrayInit(DetectEngineThreadCtx *det_ctx, uint32_t size);
 void RuleMatchCandidateTxArrayFree(DetectEngineThreadCtx *det_ctx);
 
-void AlertQueueInit(DetectEngineThreadCtx *det_ctx);
-void AlertQueueFree(DetectEngineThreadCtx *det_ctx);
 void AlertQueueAppend(DetectEngineThreadCtx *det_ctx, const Signature *s, Packet *p, uint64_t tx_id,
```
Comment (inashivb) [inline src/detect.h:1597]: I think this fn is also duplicated.

**Comment by Nancyenos** [comment]
Hey @inashivb , please can you kindly explain what am doing wrong with the commit? 
I think am not getting it
Thank you

**Comment by inashivb** [comment]
> Hey @inashivb , please can you kindly explain what am doing wrong with the commit? I think am not getting it Thank you

Just add a prefix `misc: `  to the subject like https://github.com/OISF/suricata/commit/c9e4524b3c71d0177 😉

**Comment by Nancyenos** [comment]
> > Hey @inashivb , please can you kindly explain what am doing wrong with the commit? I think am not getting it Thank you
> 
> Just add a prefix `misc: ` to the subject like [c9e4524b3c71d0177](https://github.com/OISF/suricata/commit/c9e4524b3c71d0177) 😉

ooh. Thank you/Totally missed it :smiley:

**Comment by Nancyenos** [comment]
work continued in #11998

**Pull request closed without merging**